presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveCoercion.java

```diff
@@ -383,6 +391,8 @@ private void doTestHiveCoercion(HiveTableDefinition tableDefinition)
                         //new BigDecimal("-12345.12345"),
                         -Float.parseFloat(decimalToFloatVal),
                         -12345.12345,
+                        "abc",
```

Comment:
Add a test case where eg `💰💰💰` (varchar(3)) gets truncated to varchar(2) resulting in `💰💰`

Reply:
the last value is truncating from `abc` (varchar(3)) to `ab` (varchar(2)). Could you please elaborate on what you're suggesting here?

Reply:
The point is -- see https://github.com/prestosql/presto/pull/5530/files#r506265902
You are truncating using `String#substring`, which is correct for values within Unicode BMP, but not correct for characters encoded in UTF-16 as surrogate pairs. 💰 is an example such character. Current code will probably truncate to `💰` instead of `💰💰`.

Reply:
Got it, will look into it. thanks for the explanation.